moshi/src/main/java/com/kroger/moshi/MoshiCacheSerializer.kt

```diff
@@ -0,0 +1,42 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2023 The Kroger Co. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.kroger.moshi
+
+import com.kroger.cache.CacheSerializer
+import com.squareup.moshi.JsonAdapter
+import javax.inject.Inject
+
+public class MoshiCacheSerializer<T> @Inject constructor(private val adapter: JsonAdapter<T>) : CacheSerializer<T> {
+    override fun decodeFromString(bytes: ByteArray?): T? = if (bytes == null || bytes.isEmpty()) {
```

Comment:
Should this function be called `decodeFromByteArray` instead?

Reply:
We could rename it that, but that would technically be a breaking change.

Reply:
maybe a comment acknowledging and explaining the disconnect in the function name and type then?

Reply:
I'm ok with the breaking change to correct it now.